src/entrypoint.rs

```diff
@@ -82,7 +82,11 @@ impl Drop for RunPg {
     }
 }
 
-pub async fn common_main(config: conf::Conf, crypto: Option<SharedBlstCrypto>) -> Result<()> {
+pub async fn common_main(
```

Comment:
instead of a boolean paramter, I would more have two separate functions: e.g. main_process() & main_loop() 

with same body, but different end handler.exit_loop() or handler.exit_process

There are a lot of articles out there to explain why boolean parameters should be avoided when possible